From e90ae8cee61028edbe859363a55138d1488c0efb Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Fri, 23 Feb 2018 10:29:00 +0100
Subject: [PATCH 2/2] debuild: only set --rootcmd from an user explicit choice

In the absence of an explicit user choice, no -r option was generated
with --no-conf, and -rfakeroot without --noconf. This inconsistency
was not a problem, as dpkg-buildpackage defaults to fakeroot.

The switch to confvar has (unwantedly) changed this, making debuild
pass -rfakeroot in both cases. Now test_package_lifecycle fails
because it sees an unexpected -rfakeroot in the log.

This patch lets debuild only overrides dpkg default if an user option
asks for it, in both cases. IMHO this has always been the intended
behaviour :-).
---
 scripts/debuild.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/debuild.pl b/scripts/debuild.pl
index 3569ccb7..03f4d3b7 100755
--- a/scripts/debuild.pl
+++ b/scripts/debuild.pl
@@ -174,7 +174,7 @@ EOF
 my $preserve_env=0;
 my $preserve_envvars = '';
 my %save_vars;
-my $root_command='fakeroot';
+my $root_command = undef; # Meaning no -r option, and dpkg uses fakeroot.
 my $run_lintian=1;
 my $lintian_exists=0;
 my @dpkg_extra_opts=();
-- 
2.16.1

